### PR TITLE
native/nasm: Update native nasm from verstion 2.14.02 to 2.15.05

### PR DESCRIPTION
**Pull request by th0ma7**
_Motivation:_  While working on fixing parallel build (PR #4689) I came across an build issue with nasm (caused by the parallel build and not nasm itself, and fixed with https://github.com/SynoCommunity/spksrc/pull/4689/commits/0484c441bea5f9967e26da51f75c979ca39fab0f).  While further looking into things I tested updating to latest version which worked out all right.
_Linked issues:_  https://github.com/SynoCommunity/spksrc/pull/4689/commits/0484c441bea5f9967e26da51f75c979ca39fab0f

### Checklist
- [ ] Build rule `all-supported` completed successfully
- [ ] Package upgrade completed successfully
- [ ] New installation of package completed successfully

### CODE REVIEW

**Comment by hgy59** [comment]
I have just started to redesign native packages to use installers for amd64/x86_64 target instead of compiling from sources.

So I would like to hold this back for a moment.

**Comment by hgy59** [comment]
I just found that there exists no native nasm linux installer package.
So either we use deb package to install such native packages or keep to build from source. 😞

File: native/nasm/Makefile [hunk 1]
```diff
@@ -1,8 +1,8 @@
 PKG_NAME = nasm
-PKG_VERS = 2.14.02
+PKG_VERS = 2.15.05
 PKG_EXT = tar.gz
 PKG_DIST_NAME = $(PKG_NAME)-$(PKG_VERS).$(PKG_EXT)
-PKG_DIST_SITE = https://www.nasm.us/pub/nasm/releasebuilds/$(PKG_VERS)
+PKG_DIST_SITE = https://www.nasm.us/pub/$(PKG_NAME)/releasebuilds/$(PKG_VERS)
```
Comment (hgy59) [inline native/nasm/Makefile:5]: _I am not a fan of using $(PKG_NAME) in url, as you cannot doubleclick the link. I know the links are invalid with $(PKG_VERS) anyway, but $(PKG_VERS) is variable, but $(PKG_NAME) is const within a package._ Just my own opignion.

**Review by hgy59 — APPROVED** [review]
as there are only deb packages for installation, let's keep building this from source.

File: native/nasm/Makefile (hunk 1 above)
Comment (th0ma7) [reply native/nasm/Makefile]: I am of the exact same opinion.  That specific change now reverted.

**Pull request merged**